nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -1584,7 +1584,7 @@ public DocumentModel saveDocument(DocumentModel docModel) {
             // - previous document is checkouted
             // - we don't ask for a version without updating the document (manual versioning only)
             // no need to fire event, as we use DocumentModel API it's already done
-            if (previousDocModel.isCheckedOut() && (!manualVersioning || dirty)) {
+            if (!docModel.isRecord() && docModel.isCheckedOut() && (!manualVersioning || dirty)) {
```

Comment:
Shouldn't be the following?
```suggestion
            if (!docModel.isRecord() && previousDocModel.isCheckedOut() && (!manualVersioning || dirty)) {
```

Reply:
absolutely